mlflow/store/artifact/r2_artifact_repo.py

```diff
@@ -30,5 +30,7 @@ def parse_s3_compliant_uri(self, uri):
             path = path[1:]
         return bucket, path
 
-    def _get_s3_client(self, addressing_style="virtual"):
-        return super()._get_s3_client(addressing_style=addressing_style)
+    def _get_s3_client(self, addressing_style="virtual", s3_endpoint_url=None):
+        # setup Cloudflare R2 backend to be endpoint_url, otherwise all s3 requests will go to AWS S3 by default
+        s3_endpoint_url = f"https://{urllib.parse.urlparse(self.artifact_uri).netloc.split('@')[-1]}"
```

Comment:
Super nit: it might help to break this up a bit for debuggability / readability. Also curious why we need to split on `@` manually instead of it being parsed naturally by `urllib`, is the UC Cloudflare URI format nonstandard somehow?

Also separately - we should add a unit test(s) covering the new logic, which might also help clarify how the code here works

Reply:
The UC Cloudflare URI format supported is like 
`r2://<bucket-name>@<account-id>.r2.cloudflarestorage.com/<path>`
The s3_endpoint_url however would look like https://<account-id>.r2.cloudflarestorage.com.

updated to break the parse steps and unit tests to clarify what the parse does.